client/app/pages/queries/hooks/useAutoLimitFlags.js

```diff
@@ -0,0 +1,24 @@
+import { useCallback, useState } from "react";
+import localOptions from "@/lib/localOptions";
+import { get, extend } from "lodash";
+
+function isAutoLimitAvailable(dataSource) {
+  return get(dataSource, "apply_auto_limit", false);
+}
+
+export default function useAutoLimitFlags(dataSource, query, setQuery) {
+  const isAvailable = isAutoLimitAvailable(dataSource);
+  const [isChecked, setIsChecked] = useState(localOptions.get("apply_auto_limit", true));
+  query.options.apply_auto_limit = isAvailable && isChecked;
+
+  const setAutoLimit = useCallback(
+    state => {
+      setIsChecked(state);
+      localOptions.set("apply_auto_limit", state);
```

Comment:
Why the switch to snake case?

Reply:
Because we did not have parsing when save the query. So to keep it consistent with the query execution, we have to either add parsing when saving, or change it to snake case. In order to keep things simple, I think it's better to just change it to snake case instead of adding a separate parsing.

Reply:
Oh for this one yes, but since I changed the `query.options.applyAutoLimit` to `query.options.apply_auto_limit`because of the reason I said above, I feel like it's better to keep them consistent, so I also renamed this one.

Reply:
I think it's not a problem to have it as `query.options.applyAutoLimit`, overall it's common to use the camel case for options (e.g.: for widgets and visualizations). But it would be better to have the `localOption` as `applyAutoLimit` regardless as it is already a standard for other local storage options.

Reply:
Sure, I can change this localOption to `applyAutoLimit` if you think that's better

Reply:
Let's change it only in `localOptions` because it's a client side thing (and to be consistent with the other settings we have there). We already have these differences between objects from backend and objects from frontend in other places.